beacon-chain/rpc/nodev1/node.go

```diff
@@ -264,13 +269,17 @@ func (ns *Server) GetHealth(ctx context.Context, _ *emptypb.Empty) (*emptypb.Emp
 	ctx, span := trace.StartSpan(ctx, "nodev1.GetHealth")
 	defer span.End()
 
+	if ns.SyncChecker.Synced() {
+		return &emptypb.Empty{}, nil
+	}
 	if ns.SyncChecker.Syncing() || ns.SyncChecker.Initialized() {
+		_ = grpc.SetHeader(ctx, metadata.Pairs(grpcutils.HttpCodeMetadataKey, strconv.Itoa(http.StatusPartialContent)))
 		return &emptypb.Empty{}, nil
```

Comment:
I decided to return a positive result here. Added a comment explaining why.
> We return a positive result because failing to set a non-gRPC related header should not cause the gRPC call to fail.